promql/engine.go

```diff
@@ -2583,46 +2607,77 @@ func (ev *evaluator) aggregation(op parser.ItemType, grouping []string, without
 		case parser.SUM:
 			if s.H != nil {
 				group.hasHistogram = true
+				if group.hasFloat && group.hasHistogram {
+					break
```

Comment:
I'm not sure if that additional check here is really helping. The case of mixed aggregations should be very rare. By adding this check here, we are creating overhead for the common case and get just a bit of performance back for the rare error case.